api/spec/spec_handler.go

```diff
@@ -847,3 +859,353 @@ func (d *specHandler) SpecFromString(
+			} else {
+				return nil, nil, fmt.Errorf("invalid sharedv4 service type: %v", v)
+			}
+		case api.SpecSharedv4ExternalAccess:
+			if spec.Sharedv4ServiceSpec == nil {
+				spec.Sharedv4ServiceSpec = &api.Sharedv4ServiceSpec{}
+			}
+			if externalAccess, err := strconv.ParseBool(v); err != nil {
+				return nil, nil, err
+			} else {
+				spec.Sharedv4ServiceSpec.ExternalAccess = externalAccess
+			}
+			if spec.Sharedv4ServiceSpec.ExternalAccess {
+				if spec.Sharedv4ServiceSpec.Type != api.Sharedv4ServiceSpec_NODEPORT && spec.Sharedv4ServiceSpec.Type != api.Sharedv4ServiceSpec_LOADBALANCER {
+					return nil, nil, fmt.Errorf("sharedv4 service type %v doesn't support external access", spec.Sharedv4ServiceSpec.Type)
+				}
+			}
+		case api.SpecSharedv4ServiceName:
+			if spec.Sharedv4ServiceSpec == nil {
+				spec.Sharedv4ServiceSpec = &api.Sharedv4ServiceSpec{}
+			}
+			spec.Sharedv4ServiceSpec.Name = v
+		case api.SpecSharedv4FailoverStrategy:
+			if spec.Sharedv4Spec == nil {
+				spec.Sharedv4Spec = &api.Sharedv4Spec{}
+			}
+			if v == api.SpecSharedv4FailoverStrategyAggressive {
+				spec.Sharedv4Spec.FailoverStrategy = api.Sharedv4FailoverStrategy_AGGRESSIVE
+			} else if v == api.SpecSharedv4FailoverStrategyNormal {
+				spec.Sharedv4Spec.FailoverStrategy = api.Sharedv4FailoverStrategy_NORMAL
+			} else if v == api.SpecSharedv4FailoverStrategyUnspecified {
+				spec.Sharedv4Spec.FailoverStrategy = api.Sharedv4FailoverStrategy_UNSPECIFIED
+			} else {
+				return nil, nil, fmt.Errorf("invalid sharedv4 failover strategy: %v", v)
+			}
+		case api.SpecMountOptions:
+			if spec.MountOptions == nil {
+				spec.MountOptions = &api.MountOptions{
+					Options: make(map[string]string),
+				}
+			}
+			options, err := parser.LabelsFromString(v)
+			if err != nil {
+				return nil, nil, fmt.Errorf("invalid mount options format %v", v)
+			}
+			spec.MountOptions.Options = options
+		case api.SpecSharedv4MountOptions:
+			if spec.Sharedv4MountOptions == nil {
+				spec.Sharedv4MountOptions = &api.MountOptions{
+					Options: make(map[string]string),
+				}
+			}
+			options, err := parser.LabelsFromString(v)
+			if err != nil {
+				return nil, nil, fmt.Errorf("invalid sharedv4 client mount options format %v", v)
+			}
+			spec.Sharedv4MountOptions.Options = options
+		case api.SpecDirectIo:
+			if directIo, err := strconv.ParseBool(v); err != nil {
+				return nil, nil, err
+			} else {
+				if spec.IoStrategy == nil {
+					spec.IoStrategy = &api.IoStrategy{}
+				}
+				spec.IoStrategy.DirectIo = directIo
+			}
+		case api.SpecProxyWrite:
+			if proxyWrite, err := strconv.ParseBool(v); err != nil {
+				return nil, nil, err
+			} else {
+				spec.ProxyWrite = convertToRestoreParamBoolType(proxyWrite)
+			}
+		case api.SpecFastpath:
+			if fastpath, err := strconv.ParseBool(v); err != nil {
+				return nil, nil, err
+			} else {
+				spec.FpPreference = convertToRestoreParamBoolType(fastpath)
```

Comment:
If an option is not specified in storage-class, (specifically the boolean values), will it make sense to have the  default value be to set to RestoreParamType_BKUPSRC?